src/Root.cpp

```diff
@@ -106,11 +106,16 @@ void cRoot::Start(std::unique_ptr<cSettingsRepositoryInterface> overridesRepo)
 	EnableMenuItem(hmenu, SC_CLOSE, MF_GRAYED);  // Disable close button when starting up; it causes problems with our CTRL-CLOSE handling
 	#endif
 
-	cLogger::cListener * consoleLogListener = MakeConsoleListener();
+	cLogger::cListener * consoleLogListener = nullptr;
 	cLogger::cListener * fileLogListener = new cFileListener();
-	cLogger::GetInstance().AttachListener(consoleLogListener);
 	cLogger::GetInstance().AttachListener(fileLogListener);
 
+	if (!m_RunAsService)
+	{
```

Comment:
Seems like the cleanest way. I've looked into ways of checking if stdout is pointing to a valid handle, but it's all messy and not very cross-platform.